Error while reading files, observing MalformedInputExceptions in SI 1.1.0

### Description

Observe java.nio.charset.MalformedInputException: Input length = 1 exceptions in SI 1.1.0 for files when reading from file source

### Steps to Reproduce

Deploy the following siddhi application
```

@App:name("SampleSiddhiApp")

@App:description("Description of the plan")

@source(type='file', mode='LINE',
    dir.uri='file:/Users/wso2/Desktop/samplefiles',
    tailing='false',
    header.present='false',
    read.only.trailer = "false",
    @map(
        type='csv',
        delimiter='|',
        @attributes(field = "0",field2 ="1", filePath = 'trp:file.path', eof = 'trp:eof')))
define stream inputstream (field string,field2 string, filePath string, eof string);

@sink(type='log') 
define stream outlogstream (field string,field2 string, filePath string, eof string);

from inputstream
select *
insert into outlogstream;
```

Add files which contain non-ascii characters  :/Users/wso2/Desktop/samplefiles directory and observe the following 
```

 ERROR {io.siddhi.extension.io.file.listeners.FileSystemListener} - Failed to provide file 'file:///Users/wso2/Desktop/samplefiles/samplefile.txt for consuming. org.wso2.carbon.messaging.exceptions.ClientConnectorException: Exception occurred while processing file: Input length = 1
.wso2.transport.remotefilesystem.server.connector.contractimpl.RemoteFileSystemServerConnectorImpl.poll(RemoteFileSystemServerConnectorImpl.java:56)
	at io.siddhi.extension.io.file.FileSourcePoller.run(FileSourcePoller.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

Caused by: java.nio.charset.MalformedInputException: Input length = 1
	at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
```


### Affected Component

SI

### Version

1.1.0

### Environment Details (with versions)

_No response_

### Relevant Log Output

_No response_

### Related Issues

_No response_

### Suggested Labels

_No response_

--- a/components/org.wso2.transport.file/src/main/java/org/wso2/transport/file/connector/sender/VFSClientConnector.java
+++ b/components/org.wso2.transport.file/src/main/java/org/wso2/transport/file/connector/sender/VFSClientConnector.java
@@ -265,18 +265,15 @@ public class VFSClientConnector implements ClientConnector {
                         String scheme = UriParser.extractScheme(fileURI);
                         String filePath;
                         String fileExtension;
+                        filePath = path.getName().getPath();
                         if (scheme == null || scheme.equals(Constants.PROTOCOL_FILE)) {
-                            filePath = path.getName().getPath();
                             fileExtension = filePath.substring(filePath.lastIndexOf(".") + 1);
-                            bufferedReader = new BufferedReader(
-                                    new InputStreamReader(path.getContent().getInputStream(), StandardCharsets.UTF_8));
                         } else {
                             String fileName = path.getName().getBaseName();
-                            filePath = path.getName().getPath();
                             fileExtension = fileName.substring(fileName.lastIndexOf(".") + 1);
-                            bufferedReader = new BufferedReader(
-                                    new InputStreamReader(path.getContent().getInputStream(), StandardCharsets.UTF_8));
                         }
+                        bufferedReader = new BufferedReader(
+                                new InputStreamReader(path.getContent().getInputStream(), StandardCharsets.UTF_8));
 
                         String mode = map.get(Constants.MODE);
                         if (Constants.MODE_TYPE_LINE.equalsIgnoreCase(mode) &&

--- a/components/org.wso2.transport.file/src/main/java/org/wso2/transport/file/connector/sender/VFSClientConnector.java
+++ b/components/org.wso2.transport.file/src/main/java/org/wso2/transport/file/connector/sender/VFSClientConnector.java
@@ -268,7 +268,8 @@ public class VFSClientConnector implements ClientConnector {
                         if (scheme == null || scheme.equals(Constants.PROTOCOL_FILE)) {
                             filePath = path.getName().getPath();
                             fileExtension = filePath.substring(filePath.lastIndexOf(".") + 1);
-                            bufferedReader = Files.newBufferedReader(Paths.get(filePath));
+                            bufferedReader = new BufferedReader(
+                                    new InputStreamReader(path.getContent().getInputStream(), StandardCharsets.UTF_8));
                         } else {
                             String fileName = path.getName().getBaseName();
                             filePath = path.getName().getPath();

